airflow/www/api/experimental/endpoints.py

```diff
@@ -84,8 +90,11 @@ def trigger_dag(dag_id):
 
     if getattr(g, 'user', None):
         _log.info("User {} created {}".format(g.user, dr))
-
-    response = jsonify(message="Created {}".format(dr))
+    if detailed_output:
+        dr = format_dag_run(dr)
+        response = jsonify(**dr)
+    else:
+        response = jsonify(message="Created {}".format(dr))
```

Comment:
Why  not have both?

```
return  jsonify(message="Created {}".format(dr), **dr)
```

That makes it easier to deprecate  the old message over time without also having to add special handling of the new parameter.